server/utils/cloudFunctions/functions/test/countAdvertisersTests/countAdvertisersFunctions.test.js

```diff
@@ -0,0 +1,406 @@
+/**
+ * Description: Unit tests for countAdvertisers.js
+ *              Uses sinon/chai to spy on calls and validate behavior.
+ *              compile with `npm run test`.
+ * Tests are structured in the following format: 
+ * - Setup snapshot data.
+ * - Initialize wrapper(s).
+ * - Combine wrapper(s) with snapshot, creating a promise.
+ *   - Promises must return something or catch an error.
+ *     - Default pattern with these tests is return undefined at the end 
+ *       of a nest, and to catch all possible errors when `getting` a doc 
+ *       from the db. 
+ *   - This is generally done by nesting promises. 
+ *   - There is a debate about nesting vs chaining promises. In this case, 
+ *     nesting is advantageous because it easily keeps everything in scope
+ *     without much fuss. Thus, nesting is fine to use. 
+ *   - Also this is how the official docs do it. 
+ * Testing note: 
+ * - Firebase recommends running tests online, i.e., on a real db
+ *   for testing simplicity.
+ *   - See: https://firebase.google.com/docs/functions/get-started
+ * - Test cleanup is performed by an `after` function, that wipes 
+ *   all touched collections. 
+ *   - Firestore doesn't have a default method to delete collections
+ *     so a method has been taken from firestore docs and is in the `test`
+ *     directory as `deleteCollection.js`.   
+ * Author: Robert Marcus
+ * Date: July 13, 2020
+ */
+
+// Import the testing environment configuration
+const { test, 
+        assert, 
+        sinon, 
+        chai,
+        firestoreMock, 
+        firestoreWrap, 
+        snapFromJson,
+        DEV_ADS_PATH } = require('../testConfig');
+
+const { deleteCollection } = require('../deleteCollection');
+
+const { DB } = require('../../firebaseConfig');
+
+const { DEV_AGGREGATES_COLLECTION } = 
+    require('../../countAdvertisersCloudFunctions/countAdvertisersConfig');
+
+// Import the cloud functions.
+const { devCountAdvertisersOnCreate, devCountAdvertisersOnDelete } = 
+    require('../../countAdvertisersCloudFunctions/devCountAdvertisersFunctions');
+
+// Save some space for test explanation in the test description
+const SHOULD_INCREMENT = "should increment advertisers ad count "; 
+
+describe('Count advertisers cloud functions', () => {
+  // After every 'describe' block, reset the test environments. 
+  // Since these are live dev db's, this process can be very flaky.
+  // In general, don't expect an 'advertiser' will be reset from 
+  // test to test. 
+  after(() => {
```

Comment:
this test can be shorter by having an array of collections `const collections = ['dev_ads', ''dev_aggregates/2018/advertisers', 'dev_aggregates/2019/advertisers', 'dev_aggregates/2020/advertisers'];` then map over the array and `deleteCollection` :D